### PR TITLE
[aspnetcore] Added parameter to generate models in a seperate csproj

### PR DESCRIPTION
**Pull request by knom**
@wing328 
### [aspnetcore] Added parameter to generate models in a seperate csproj
Now a seperate CSPROJ can be generated for MODELS.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
@knom thanks for the PR. Can you please elaborate on the use case for this option? 

cc @mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java [hunk 1]
```diff
@@ -393,17 +400,31 @@ public void processOpts() {
...
+
         if (aspnetCoreVersion.getOptValue().startsWith("3.") || aspnetCoreVersion.getOptValue().startsWith("5.0") || aspnetCoreVersion.getOptValue().startsWith("6.")) {
             supportingFiles.add(new SupportingFile("OpenApi" + File.separator + "TypeExtensions.mustache", packageFolder + File.separator + "OpenApi", "TypeExtensions.cs"));
         }
+
+        if (useSeperateModelProject)
+        {
+            supportingFiles.add(new SupportingFile("ModelsProject.csproj.mustache", sourceFolder + File.separator + modelPackage, modelPackage + ".csproj"));
+        }
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java:427]: can you please update the Java code to conform the java code style? e.g.
```java
if (...) {
    // do something
} else {
    // do something else
}
```

Comment (knom) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java:426]: Fine @wing328 I've refactored the IF/ELSEs

**Comment by knom** [comment]
> @knom thanks for the PR. Can you please elaborate on the use case for this option?
> 
> cc @mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02) @Blackclaws (2021/03) @lucamazzanti (2021/05)

In C# some folks like to have MODELS in a seperate C# project from the CONTROLLERS and stuff.
That is for better structuring - and reuse of the models.

The new switch will generate 2 projects:
* 1 default project (Controllers, startup.cs, ...)
* 1 project just for models

**Comment by wing328** [comment]
Thanks for the explanation. Let's give it a try.

**Comment by knom** [comment]
I guess we'd probably need special sample generators and CI test..

Not sure how to do those.. can you help me?

**Pull request merged**